
This is mainly needed when used with dietlibc - but also more portable
and more secure.

  Rene Rebe <rene@rocklinux.org>

diff -ur kiss-0.21/src/doread.c src.kiss.-1.25906.3396530713/kiss-0.21/src/doread.c
--- kiss-0.21/doread.c	1998-09-25 14:07:09.000000000 +0200
+++ kiss-0.21/doread.c	2003-08-17 23:54:10.000000000 +0200
@@ -17,11 +17,11 @@
 	for (cp = s.str [1]; *cp; cp++)
 	    if (! isupper (*cp))
 		return (warning ("variable must be in upper case"));
-	gets (buf);
+	fgets (buf, FILENAMELEN, stdin);
 	addtoenv (s.str [1], buf);
     }
     else
-	gets (buf);
+	fgets (buf, FILENAMELEN, stdin);
     
     return (0);
 }
diff -ur kiss-0.21/src/runcmd.c src.kiss.-1.25906.3396530713/kiss-0.21/src/runcmd.c
--- kiss-0.21/runcmd.c	1998-09-25 14:07:10.000000000 +0200
+++ kiss-0.21/runcmd.c	2003-08-17 23:11:35.000000000 +0200
@@ -1,5 +1,7 @@
 #include "kiss.h"
 
+#include <getopt.h>
+
 /* define DEBUG_ONE_CMD to get one shot at gdb'ing the child process */
 /* then make a -g version of the command to see and set a trace in the code */
 
